test/ad/distributions.jl

```diff
@@ -73,8 +73,8 @@
         DistSpec(Beta, (1.0, 2.0), 0.5),
 
         DistSpec(BetaPrime, (), 0.5),
-        DistSpec(BetaPrime, (1.0,), 0.5),
-        DistSpec(BetaPrime, (1.0, 2.0), 0.5),
+        DistSpec(BetaPrime, (1.5,), 0.5),
```

Comment:
I also noticed the same issue but thought it only affects ReverseDiff with tape caching. When tape caching is disabled, ReverseDiff should handle control flows correctly (similar to Tracker).

Reply:
I don't think it's ReverseDiff specific since e.g. Tracker failed as well (https://github.com/TuringLang/DistributionsAD.jl/pull/185/checks?check_run_id=3338494448#step:5:1097). In fact all AD systems apart from Zygote return a value that is different from finite differencing:
```julia
julia> using Distributions, FiniteDifferences, Tracker, ReverseDiff, ForwardDiff, Zygote

julia> f(x) = logpdf(BetaPrime(x), 0.5)
f (generic function with 1 method)

julia> central_fdm(5, 1)(f, 1.0)
0.495922603223565

julia> central_fdm(12, 1)(f, 1.0)
0.4959226032237455

julia> Tracker.gradient(f, 1.0)
(1.1890697837836721 (tracked),)

julia> ForwardDiff.derivative(f, 1.0)
1.1890697837836721

julia> ReverseDiff.gradient(f ∘ first, [1.0])
1-element Vector{Float64}:
 1.1890697837836721

julia> Zygote.gradient(f, 1.0)
(0.49592260322372683,)
```

The different values arise from the different behaviour of the AD systems for `x -> xlogy(x, 0.5)` at `0`. FiniteDifferences and Zygote return the correct derivative (`log(0.5)`) whereas Tracker, ForwardDiff, and ReverseDiff return 0:
```julia
julia> using LogExpFunctions

julia> central_fdm(12, 1)(x -> xlogy(x, 0.5), 0.0)
-0.6931471805599452

julia> Tracker.gradient(x -> xlogy(x, 0.5), 0.0)
(0.0 (tracked),)

julia> ForwardDiff.derivative(x -> xlogy(x, 0.5), 0.0)
0.0

julia> ReverseDiff.gradient(x -> xlogy(first(x), 0.5), [0.0])
1-element Vector{Float64}:
 0.0

julia> Zygote.gradient(x -> xlogy(x, 0.5), 0.0)
(-0.6931471805599453,)
```

It seems the AD systems handle the branch in https://github.com/JuliaStats/LogExpFunctions.jl/blob/39223ba1daa0244fae1885cd0bcba5a25743349b/src/basicfuns.jl#L29 differently/incorrectly.